### PR TITLE
[foreman_openscap] Add foreman OpenSCAP client plugin

### PR DESCRIPTION
**Pull request by pmoravec**
Collect /etc/foreman_scap_client/config.yaml on any Foreman OpenSCAP client system.

Resolves: #3115

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Review by TurboTurtle — APPROVED** [review]
Ack, but is there really nothing else to collect here?

**Comment by TurboTurtle** [comment]
F35 build is failing due to F35 being EOL and Fedora mirror manager not updating for it. Fix in #3114 (pending merge).

**Comment by pmoravec** [comment]
Yes, really just the one file, see https://bugzilla.redhat.com/show_bug.cgi?id=2160823#c2 .

**Comment by pmoravec** [comment]
> Yes, really just the one file, see https://bugzilla.redhat.com/show_bug.cgi?id=2160823#c2 .

Well, I am not happy to have such a "one file plugin". Maybe https://github.com/theforeman/foreman_scap_client can be extended by a file `/etc/sos/extras.d/foreman_scap_client` that will contain `:/etc/foreman_scap_client/config.yaml` to let `sos_extras` plugin to collect the config? I will check with them.

**Comment by TurboTurtle** [comment]
Either way works for me. Single file plugins aren't fantastic, but they aren't unprecedented in the project either. I'll hold off merging until we hear back on what the foreman project says about including an `sos_extras` file.

**Comment by evgeni** [comment]
> > > Yes, really just the one file, see https://bugzilla.redhat.com/show_bug.cgi?id=2160823#c2 .
> > 
> > 
> > Well, I am not happy to have such a "one file plugin". Maybe https://github.com/theforeman/foreman_scap_client can be extended by a file `/etc/sos/extras.d/foreman_scap_client` that will contain `:/etc/foreman_scap_client/config.yaml` to let `sos_extras` plugin to collect the config? I will check with them.
> 
> @ehelms, @ares, @evgeni - do you see such enhancement possible on `foreman_scap_client`, which would newly deliver (or deploy in post-install script) one file to let the client's config be automatically collected by sosreport?

There is a convention that if a directory is to be used by multiple packages, but should not be owned by them, to introduce a `filesystem` package:

```console
 % rpm -ql vim-filesystem-9.0.1160-1.fc37.noarch 
/usr/lib/rpm/macros.d/macros.vim
/usr/share/vim/vim90/import
/usr/share/vim/vim90/import/dist
/usr/share/vim/vimfiles/after
/usr/share/vim/vimfiles/after/autoload
/usr/share/vim/vimfiles/after/colors
/usr/share/vim/vimfiles/after/compiler
…
```

As there is no such package, working around "shipping but not owning" seems rather too complicated for the scap_client package.

> Since we generally dont know if the SCAP client has `sos` installed, deploying the file would have to be conditionaly only when `/etc/sos/extras.d` directory exists (or it would have to create (but not own) the directory).
> 
> This obstacle makes me feel we should stick with the `sos` tiny plugin, rather..?

I'd prefer the tiny plugin, yes.

**Comment by TurboTurtle** [comment]
Ack to tiny plugin, then. #3114 has been merged, so a rebase on `main` will resolve the CI build issue.

File: sos/report/plugins/foreman_openscap.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class ForemanOpenSCAP(Plugin, IndependentPlugin):
+
+    short_desc = 'Foreman OpenSCAP client'
+
+    plugin_name = 'foreman_openscap'
+    profiles = ('sysmgmt',)
+
+    packages = ('rubygem-foreman_scap_client',)
```
[the remark this answers is not in the document]
Comment (pmoravec) [reply sos/report/plugins/foreman_openscap.py:21]: .. or add Ubuntu package name which should be `foreman_scap_client`, if I am not mistaken..?

[the remark this answers is not in the document]
Comment (evgeni) [reply sos/report/plugins/foreman_openscap.py:21]: `ruby-foreman-scap-client`

**Pull request merged**